### ISSUE TITLE
Text in drawings is not exported to Therion format

### ISSUE THREAD

**Issue by CaverBruce** [richsmith/sexytopo#56]
All data collected and created with ST should be exported.
However text point entries in drawings is not exported to Therion format.  Not sure how this would be done, as I assume there is no direct equivalent in xvi format (I don't know).  The text entity could be parsed direct to a scrap (with point label) in the Therion th2 file, but this would be a bit messy.  Firstly all the labels would be in one arbitrary scrap, which may have no relation to the way the survey/drawing author wants to set things out, and it may confuse inexperienced Therion users.
But mainly I don't like the 'raw survey notes in scrap' idea because it blurs the line between survey notes and subsequent drawing or data edits (which is one reason why I believe TopoDROID is flawed). The ideal would be to export the ST drawing text entities into xvi format and have it parse to Therion (and other export formats).

**Comment by richsmith** [comment]
Yes, this is a bit of an issue (probably not critical as you can just refer to your survey for now).

I think you've covered the possible options. Exporting to XVI format would work but I'm not sure how to do it. I'm not sure if there's any library out there that can turn text into some simplified paths. I had a quick look but couldn't see anything obvious.

Exporting to a Therion label is easy but flawed.

Not quite sure what I'm going to do with this yet...

**Comment by CaverBruce** [comment]
How about proposing to the Therion authors that the xvi format be extended to allow text points, rather than creating stroked line paths?
From looking at xvi files created from PocketTopo, they seem to include;
XVIgrids = spacings
XVIstations = projected coordinates followed by station number (and XTherion displays the station number in the status bar if the mouse pointer is hovered over the point)
XVIshots = projected coordinate pairs for station lines
XVIsketchlines = colours followed by coordinate sets
XVIgrid = ? size and position of grid ?
There must be something for LRUD for xvi of manual surveys (with LRUD) created with a Therion export -output cave.xvi as well, but I don't have any examples to hand.

A new type of data could be;
XVIsketchtext = colours followed by insertion point coordinates and the text string 

XTherion and anything else that depends on the xvi files would need to accommodate this change. Might be a degree of coordination required across packages.  In the case of XTherion, it could behave just like the XVIstations data - display a point, and if the mouse pointer is hovered, display the text string in the status bar.  Not super user friendly, but it works within the constraints of the existing Therion interface without complicating the data (assuming a text string is simple and compact, and a line path is obscure and relatively storage intensive).

**Comment by richsmith** [comment]
Yep, this could also be an option. Something to think about after the high priority stuff like Calibration is done.

**Comment by CaverBruce** [comment]
Just bumped into this bogey again.  TopoDroid manages to plot sketch text entities as stroked line paths for export to xvi.  Perhaps when the time comes there is some code that can be borrowed...

**Comment by richsmith** [comment]
Done. Will be released in the next version.
